src/graphql/mintNfts.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  nonNull,
+  inputObjectType,
+  mutationField,
+  arg,
+  objectType,
+} from 'nexus';
+import { YogaInitialContext } from 'graphql-yoga';
+import { decryptEncodedPayload } from '../lib/cryptography/utils.js';
+
+import { Metaplex, keypairIdentity } from '@metaplex-foundation/js-next';
+import { Connection, clusterApiUrl, Keypair } from '@solana/web3.js';
+
+export const MintNftResult = objectType({
+  name: 'MintNftResult',
+  description: 'The result for minting a NFT',
+  definition (t) {
+    t.nonNull.string('message', {
+      description: 'Mint hash of newly minted NFT',
+    });
+  },
+});
+
+export const MintMetadata = inputObjectType({
+  name: 'MintMetadata',
+  description: 'The NFT metadata JSON',
+  definition (t) {
+    t.nonNull.string('metadataJson', {
```

Comment:
This shouldn't be a raw stringified JSON as it makes the api a bit harder to use.